docs/generator/generator.go

```diff
@@ -0,0 +1,307 @@
+			}
+			def := ""
+			if prop.Default != nil {
+				b, err := prop.Default.MarshalJSON()
+				handleGeneralError(fmt.Sprintf("cannot marshal property %q default value in Kamelet %s", key, k.Name), err)
+				def = "`" + strings.ReplaceAll(string(b), "`", "'") + "`"
+			}
+			ex := ""
+			if prop.Example != nil {
+				b, err := prop.Example.MarshalJSON()
+				handleGeneralError(fmt.Sprintf("cannot marshal property %q example value in Kamelet %s", key, k.Name), err)
+				ex = "`" + strings.ReplaceAll(string(b), "`", "'") + "`"
+			}
+			content += tableLine(name, prop.Title, prop.Description, prop.Type, def, ex)
+		}
+
+		content += "|===\n"
+		content += "\n"
+		content += "NOTE: Fields marked with ({empty}*) are mandatory.\n"
+
+	} else {
+		content += "The Kamelet does not specify any configuration option.\n"
+	}
+
+	content += "\n"
+	content += "== Usage\n"
+	content += "\n"
+	content += fmt.Sprintf("This section summarizes how the `%s` can be used in various contexts.\n", k.Name)
+	content += "\n"
+
+	tp := k.ObjectMeta.Labels["camel.apache.org/kamelet.type"]
+	if tp != "" {
+		content += fmt.Sprintf("=== Knative %s\n", strings.Title(tp))
+		content += "\n"
+
+		content += fmt.Sprintf("The `%s` Kamelet can be used as Knative %s by binding it to a Knative object.\n", k.Name, tp)
+		content += "\n"
+
+		sampleConfig := make([]string, 0)
+		for _, key := range keys {
+			if !required[key] {
+				continue
+			}
+			prop := k.Spec.Definition.Properties[key]
+			if prop.Default == nil {
+				ex := ""
+				if prop.Example != nil {
+					b, err := prop.Example.MarshalJSON()
+					handleGeneralError(fmt.Sprintf("cannot marshal property %q example value in Kamelet %s", key, k.Name), err)
+					ex = string(b)
+				}
+				if ex == "" {
+					ex = `"The ` + prop.Title + `"`
+				}
+				sampleConfig = append(sampleConfig, fmt.Sprintf("%s: %s", key, ex))
+			}
+		}
+		props := ""
+		if len(sampleConfig) > 0 {
+			props += "    properties:\n"
+			for _, p := range sampleConfig {
+				props += fmt.Sprintf("      %s\n", p)
+			}
+		}
+
+		kameletRef := fmt.Sprintf(`    ref:
+      kind: Kamelet
+      apiVersion: camel.apache.org/v1alpha1
+      name: %s
+%s`, k.Name, props)
+
+		knativeRef := `    ref:
+      kind: InMemoryChannel
+      apiVersion: messaging.knative.dev/v1
+      name: mychannel
+`
+
+		sourceRef := kameletRef
+		sinkRef := knativeRef
+		if tp == "sink" {
+			sourceRef = knativeRef
+			sinkRef = kameletRef
+		}
+
+		binding := fmt.Sprintf(`apiVersion: camel.apache.org/v1alpha1
+kind: KameletBinding
+metadata:
+  name: %s-binding
+spec:
+  source:
+%s sink:
+%s
+`, k.Name, sourceRef, sinkRef)
+
+		content += fmt.Sprintf(".%s-binding.yaml\n", k.Name)
+		content += "[source,yaml]\n"
+		content += "----\n"
+		content += binding
+		content += "----\n"
+
+		content += "\n"
+		content += "Make sure you have https://camel.apache.org/camel-k/latest/installation/installation.html[Camel K installed] into the Kubernetes cluster you're connected to.\n"
```

Comment:
We need a `xref` link here, otherwise link checker will not be able to verify the link and the [HTML validation](https://github.com/apache/camel-website/blob/32188349a4e0f3c411d6e97ca92b6f733d67849c/.htmlvalidate.json#L17) not allowing absolute URLs to camel.apache.org will fail.

```suggestion
		content += "Make sure you have xref:latest@camel-k::installation/installation.html[Camel K installed] into the Kubernetes cluster you're connected to.\n"
```